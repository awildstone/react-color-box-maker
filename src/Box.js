import React from 'react';

const Box = ({ id, bgColor="purple", width=100, height=100, removeBox }) => {
    const style = {
        backgroundColor: bgColor,
        width: `${width}px`,
        height: `${height}px`
    };

    const handleClick = () => {
        removeBox(id);
    }
    
    return (
        <div>
            <div style={style} data-testid={id}></div>
            <button onClick={handleClick}>X</button>
        </div>
    )
}

export default Box;