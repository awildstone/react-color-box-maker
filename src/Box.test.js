import React from 'react';
import { render } from '@testing-library/react';
import Box from './Box';

it('renders a Box without crashing', () => {
    render(<Box />);
});

it('matches a Box snapshot', () => {
    const { asFragment } = render(<Box id={1} bgColor={"#ffffff"} width={"600"} height={"500"} />);
    expect(asFragment()).toMatchSnapshot();
});