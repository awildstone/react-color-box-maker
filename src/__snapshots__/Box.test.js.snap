// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches a Box snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      data-testid="1"
      style="background-color: rgb(255, 255, 255); width: 600px; height: 500px;"
    />
    <button>
      X
    </button>
  </div>
</DocumentFragment>
`;
